robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowCompoundButton.java

```diff
@@ -0,0 +1,33 @@
+package org.robolectric.shadows;
+
+import android.graphics.drawable.Drawable;
+import android.widget.CompoundButton;
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.annotation.RealObject;
+
+@SuppressWarnings({"UnusedDeclaration"})
+@Implements(CompoundButton.class)
+public class ShadowCompoundButton extends ShadowTextView {
+  @RealObject CompoundButton realCompoundButton;
+  private int buttonDrawableId;
+  private Drawable buttonDrawable;
+
+  @Implementation
+  public void setButtonDrawable(int buttonDrawableId) {
+    this.buttonDrawableId = buttonDrawableId;
```

Comment:
It might be nice to call-through for this method (i.e. `directlyOn(realObject, CompoundButton.class, ...)` etc.).